bundles/org.openhab.binding.amazonechocontrol/src/main/java/org/openhab/binding/amazonechocontrol/internal/handler/SmartHomeDeviceHandler.java

```diff
@@ -53,28 +76,76 @@
 
     private final Logger logger = LoggerFactory.getLogger(SmartHomeDeviceHandler.class);
 
-    private @Nullable ScheduledFuture<?> updateStateJob;
     private @Nullable Connection connection;
-    private @Nullable SmartHomeDevice smartHomeDevice;
-
-    Storage<String> stateStorage;
+    private @Nullable SmartHomeBaseDevice smartHomeBaseDevice;
+    private Gson gson;
 
     @Nullable
     AccountHandler accountHandler;
     Thing thing;
 
-    public SmartHomeDeviceHandler(Thing thing, Storage<String> storage) {
+    public SmartHomeDeviceHandler(Thing thing, Gson gson) {
         super(thing);
         this.thing = thing;
-        this.stateStorage = storage;
+        this.gson = gson;
+    }
+
+    public boolean setDeviceAndUpdateThingState(AccountHandler accountHandler,
+            @Nullable SmartHomeBaseDevice smartHomeBaseDevice) {
+        if (this.accountHandler != accountHandler) {
+            this.accountHandler = accountHandler;
+        }
+        if (smartHomeBaseDevice == null) {
+            updateStatus(ThingStatus.UNKNOWN);
+            return false;
+        }
+        this.smartHomeBaseDevice = smartHomeBaseDevice;
+        updateStatus(ThingStatus.ONLINE);
+
+        Set<String> capabilities = GetCapabilities(accountHandler, smartHomeBaseDevice);
+        try {
+            if (capabilities.contains(INTERFACE_POWER)) {
```

Comment:
channels are defined now depending only on the Capability. **But this should be changed in the future, we should check the supported properties too!**